linker/Mono.Linker.Steps/MarkStep.cs

```diff
@@ -544,11 +544,14 @@ protected virtual TypeDefinition MarkType (TypeReference reference)
 
 			reference = GetOriginalType (reference);
 
+			if (reference == null)
+				return null;
```

Comment:
I suggest returning unchanged FunctionPointerType from GetOriginalType and changing this to 
```
if (reference is FunctionPointerType)`
          return null;
```